src/sumo_store.erl

```diff
@@ -403,29 +417,49 @@ handle_call(
   {reply, {OkOrError, Reply}, State#state{handler_state=NewState}}.
 
 %% @hidden
--spec handle_cast(term(), state()) ->
-  {noreply, state()} |
-  {noreply, state(), non_neg_integer()} |
-  {noreply, state(), hibernate} |
-  {stop, term(), state()}.
 handle_cast(_Msg, State) ->
   {noreply, State}.
 
 %% @hidden
--spec handle_info(term(), state()) ->
-  {noreply, state()} |
-  {noreply, state(), non_neg_integer()} |
-  {noreply, state(), hibernate} |
-  {stop, term(), state()}.
-handle_info(_Info, State) ->
-  {noreply, State}.
+handle_info(Info, #state{handler = Handler, handler_state = HState} = State) ->
+  case erlang:function_exported(Handler, handle_info, 2) of
+    true ->
+      try
```

Comment:
Nitpicking, but I thing the _proper_ way to do this would be:

```erlang
try Handler:handle_info(Info, HState) of
  Response -> handle_info_response(Response, State)
```